### PR TITLE
Add `capture_request_form_data` param to `instrument_httpx`

### PR DESCRIPTION
**Pull request by alexmojaki**

### CODE REVIEW

File: logfire/_internal/integrations/httpx.py [hunk 1]
```diff
@@ -158,39 +167,51 @@ def instrument_httpx(
...
+    if not should_capture_headers and not should_capture_json and not should_capture_form_data and not hook:
         return None
 
     def new_hook(span: Span, request: RequestInfo) -> None:
         with handle_internal_errors():
-            if should_capture_headers:
-                capture_request_headers(span, request)
-            if should_capture_json:
-                capture_request_body(span, request)
+            capture_request_stuff(request, span, should_capture_headers, should_capture_json, should_capture_form_data)
```
Comment (Kludex) [inline logfire/_internal/integrations/httpx.py]:
```suggestion
            capture_request(request, span, should_capture_headers, should_capture_json, should_capture_form_data)
```

File: logfire/_internal/integrations/httpx.py [hunk 2]
```diff
@@ -338,3 +359,34 @@ def capture_request_body(span: Span, request: RequestInfo) -> None:
...
+        httpx.Client.stream,
+        httpx.AsyncClient.request,
+        httpx.AsyncClient.stream,
+    ]
+]
+
+
+def capture_request_form_data(span: Span, request: RequestInfo) -> None:
+    content_type = cast('httpx.Headers', request.headers).get('content-type', '')
+    if content_type != 'application/x-www-form-urlencoded':
```
Comment (Kludex) [inline logfire/_internal/integrations/httpx.py:377]: and `multipart/form-data`?

File: logfire/_internal/integrations/httpx.py [hunk 3]
```diff
@@ -158,39 +167,51 @@ def instrument_httpx(
...
-                capture_request_headers(span, request)
-            if should_capture_json:
-                capture_request_body(span, request)
+            capture_request_stuff(request, span, should_capture_headers, should_capture_json, should_capture_form_data)
             await run_async_hook(hook, span, request)
 
     return new_hook
 
 
+def capture_request_stuff(
```
Comment (Kludex) [inline logfire/_internal/integrations/httpx.py]:
```suggestion
def capture_request(
```

File: logfire/_internal/integrations/httpx.py (hunk 2 above)
Comment (alexmojaki) [reply logfire/_internal/integrations/httpx.py:377]: Based on the httpx source code (encode_request -> encode_urlencoded_data) it's always this, multipart is for `files`.

**Review by Kludex — APPROVED** [review]

**Pull request merged**